File: tools/dbench/answers/go_haiku/s1/r1_orders.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	data, _ := os.ReadFile("orders.json")
	var orders []map[string]interface{}
	json.Unmarshal(data, &orders)

	totalPaid := 0.0
	openCount := 0

	for _, order := range orders {
		if status, ok := order["status"].(string); ok {
			if status == "paid" {
				if amt, ok := order["amt"].(float64); ok {
					totalPaid += amt
				}
			} else if status == "open" {
				openCount++
			}
		}
	}

	fmt.Println(totalPaid)
	fmt.Println(openCount)
}

File: tools/dbench/answers/rust_haiku/s1/r1_orders.rs
use std::fs;

fn main() {
    let content = fs::read_to_string("orders.json").unwrap_or_default();
    let mut paid_total = 0.0;
    let mut open_count = 0;

    let content = content.trim_matches(|c| c == '[' || c == ']');
    for obj in content.split("},{") {
        let obj = obj.trim_matches(|c| c == '{' || c == '}');

        let mut amt = 0.0;
        let mut status = "";

        for part in obj.split(',') {
            let part = part.trim();
            if part.starts_with("\"amt\"") {
                if let Some(val_start) = part.find(':') {
                    let val_str = part[val_start + 1..].trim();
                    amt = val_str.parse::<f64>().unwrap_or(0.0);
                }
            } else if part.starts_with("\"status\"") {
                if let Some(val_start) = part.find(':') {
                    let val_str = part[val_start + 1..].trim().trim_matches('"');
                    status = val_str;
                }
            }
        }

        if status == "paid" {
            paid_total += amt;
        } else if status == "open" {
            open_count += 1;
        }
    }

    println!("{}", paid_total);
    println!("{}", open_count);
}
